distribution/pull.go

```diff
@@ -147,7 +147,7 @@ func Pull(ctx context.Context, ref reference.Named, imagePullConfig *ImagePullCo
 			return TranslatePullError(err, ref)
 		}
 
-		imagePullConfig.ImageEventLogger(ref.String(), repoInfo.Name(), "pull")
+		imagePullConfig.ImageEventLogger(reference.FamiliarString(ref), reference.FamiliarName(repoInfo.Name), "pull")
```

Comment:
Below this there's `lastErr = fmt.Errorf("no endpoints found for %s", ref.String())` - should this also be converted to `FamiliarString` to preserve the old behavior?